projects/client/RabbitMQ.Client/src/client/api/IConnectionFactory.cs

```diff
@@ -129,13 +129,6 @@ public interface IConnectionFactory
         IConnection CreateConnection(IList<string> hostnames, String clientProvidedName);
 
         /// <summary>
-        /// Advanced option.
-        ///
-        /// What task scheduler should consumer dispatcher use.
-        /// </summary>
-        TaskScheduler TaskScheduler { get; set; }
```

Comment:
Yep I agree we're in the grey area. SemVer often is, especially when deprecating features. We'll push out an RC before so people can make their voices heard if they feel we've got something wrong or if there is a surprisingly large number of users that write their own TaskSchedulers.